lib/streamaggr/streamaggr.go

```diff
@@ -951,95 +1010,85 @@ func (a *aggregator) Push(tss []prompbmarshal.TimeSeries, matchIdxs []byte) {
 		key := bytesutil.ToUnsafeString(buf[bufLen:])
 		for _, s := range ts.Samples {
 			if math.IsNaN(s.Value) {
-				a.ignoredNaNSamples.Inc()
 				// Skip NaN values
+				a.ignoredNaNSamples.Inc()
 				continue
 			}
-			if ignoreOldSamples && s.Timestamp < minTimestamp {
-				a.ignoredOldSamples.Inc()
+			if (ignoreOldSamples || enableWindows) && s.Timestamp < minDeadline {
 				// Skip old samples outside the current aggregation interval
+				a.ignoredOldSamples.Inc()
 				continue
 			}
 			lagMsec := nowMsec - s.Timestamp
 			if lagMsec > maxLagMsec {
 				maxLagMsec = lagMsec
 			}
-			samples = append(samples, pushSample{
-				key:       key,
-				value:     s.Value,
-				timestamp: s.Timestamp,
-			})
+			if enableWindows && s.Timestamp <= cs.maxDeadline == cs.isGreen {
```

Comment:
I'm not sure that I understand the last condition. Maybe some comment is required to clarify it. 

`(s.Timestamp <= cs.maxDeadline) == cs.isGreen` 

Should it be `s.Timestamp <= cs.maxDeadline && cs.isGreen`  ?